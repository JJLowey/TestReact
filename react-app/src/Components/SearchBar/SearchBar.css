input
{
    /* Bootstrap overrides */
    padding: 16px 8% 16px 2%!important;
    border-color: #a9a9a9!important;
    border-style: solid!important;
    border-width: 1px!important;
    height: 50px!important;
    border-radius: 0!important;;
}

/* Identical to Bootstrap */
input:focus 
{
    outline-offset: -2px;
    outline: -webkit-focus-ring-color auto 5px;
}

label
{
    text-shadow: none;
    font-size: 0.875em;
}

.loading
{
    background-color: #ffffff;
    background-image: url("https://cdn2.rcstatic.com/images/site_graphics/newsite/preloader64.gif");
    background-size: 25px 25px;
    background-position: 98% 50%;
    background-repeat: no-repeat;
}