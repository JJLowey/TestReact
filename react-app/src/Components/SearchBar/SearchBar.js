import React from 'react';
import { debounce } from 'lodash';

import './SearchBar.css';

class SearchBar extends React.Component
{
    state = {term: ''};

    onSearchSubmit = (event) =>
    {
        // Stop enter button refreshing page
        event.preventDefault();    
        // Use the passed in callback function to respond to parent component   
        //this.props.onSubmit(this.state.term);
    }

    // onChange fires this event (debounced to reduce load on server)
    onSearchChange = debounce((text) =>
    {   
        this.setState({term: text});

        // Use the passed in callback function to respond to parent component  
        this.props.onChange(text);
    }, 500);

    render()
    {
        return (
            <div className="col-xs-12">
                <form onSubmit={this.onSearchSubmit} className="col-xs-12">
                    <div className="field">
                        <label htmlFor="pickUpInput">Pick-up Location</label>
                        <input className={`form-control col-xs-12 ${this.props.loading ? 'loading' : ''}`} id="pickUpInput" 
                                autoComplete="off"
                                type="text"
                                placeholder="city, airport, station, region and district..." 
                                onChange={event => this.onSearchChange(event.target.value)}/>
                    </div>
                </form>
            </div>
        );
    }
}

export default SearchBar;