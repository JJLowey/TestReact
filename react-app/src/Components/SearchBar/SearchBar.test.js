import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme';
import SearchBar from './SearchBar';

// Will not recognise setupTests.js :(
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('Has not changed?', () => 
{
  // Check if the component has changed? Snapshot is deep, not shallow
  const tree = renderer.create(<SearchBar/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check SearchBar Component Renders', () => 
{
  const div = document.createElement('div');
  ReactDOM.render(<SearchBar />, div);
});

describe('Testing SearchBar Component', () => 
{
    it('Should be a label for the search', () => 
    {
      const rootComponent = shallow(<SearchBar/>)
      expect(rootComponent.find('label')).toBeTruthy();
    })

    it('Label should equal "Pick-up Location"', () => 
    {
        const rootComponent = shallow(<SearchBar/>);
        expect(rootComponent.find('label').text()).toEqual('Pick-up Location');
    })

    it('Placeholder of input box should equal "city, airport, station, region and district..."', () => 
    {
        const rootComponent = shallow(<SearchBar/>);
        expect(rootComponent.find('input').prop('placeholder')).toEqual('city, airport, station, region and district...');
    })

    it('Make sure input has set the value on change"', () => 
    {

    })

})