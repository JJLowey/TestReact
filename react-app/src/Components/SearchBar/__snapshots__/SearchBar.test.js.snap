// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has not changed? 1`] = `
<div
  className="col-xs-12"
>
  <form
    className="col-xs-12"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="pickUpInput"
      >
        Pick-up Location
      </label>
      <input
        autoComplete="off"
        className="form-control col-xs-12 "
        id="pickUpInput"
        onChange={[Function]}
        placeholder="city, airport, station, region and district..."
        type="text"
      />
    </div>
  </form>
</div>
`;
